source/Applications/Advanced/HandEyeCalibration/UtilizeEyeInHandCalibration/UtilizeEyeInHandCalibration.cs

```diff
@@ -0,0 +1,157 @@
+            var endEffectorPoseFile = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData)
+                               + "/Zivid/RobotTransform.yaml";
+
+            var eyeInHandTransformation = readTransform(eyeInHandTransformationFile);
+            var endEffectorPose = readTransform(endEffectorPoseFile);
+
+            // Converting to Mat.NET matrices for easier computation
+            Matrix<float> eyeInHandMatrix = zividToMathDotNet(eyeInHandTransformation);
+            Matrix<float> endEffectorMatrix = zividToMathDotNet(endEffectorPose);
+
+            var dataFile = Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData) + "/Zivid/ZividGem.zdf";
+            Console.WriteLine("Reading ZDF frame from file: " + dataFile);
+            var frame = new Zivid.NET.Frame(dataFile);
+            var pointCloud = frame.PointCloud;
+
+            Console.WriteLine("Computing camera pose in robot base frame");
+            Matrix<float> transformBaseToCamera = endEffectorMatrix * eyeInHandMatrix;
+
+
+            switch (Interaction.EnterCommand())
+            {
+                case CommandType.TransformSinglePoint:
+
+                    Console.WriteLine("Transforming single point");
+
+                    // The (picking) point is defined as image coordinates in camera frame. It is hard-coded for the
+                    // ZividGem.zdf
+                    const int imageCoordinateX = 1357;
+                    const int imageCoordinateY = 666;
+                    var xyz = pointCloud.CopyPointsXYZW();
+
+                    var pickingPoint = new float[]{xyz[imageCoordinateY, imageCoordinateX, 0],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 1],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 2],
+                                                            xyz[imageCoordinateY, imageCoordinateX, 3]};
+
+                    Vector<float> pointInCameraFrame = CreateVector.DenseOfArray(pickingPoint);
+                    Console.WriteLine("Point coordinates in camera frame: " + pointInCameraFrame);
+
+                    Console.WriteLine("Transforming (picking) point from camera to robot base frame");
+                    Vector<float> pointInBaseFrame = transformBaseToCamera * pointInCameraFrame;
+
+                    Console.WriteLine("Point coordinates in robot base frame: " + pointInBaseFrame);
+                    break;
+
+                case CommandType.TransformPointCloud:
+
+                    Console.WriteLine("Transforming point cloud");
+                    var transformMatrix = mathDotNetToZivid(transformBaseToCamera);
+                    pointCloud.Transform(transformMatrix);
+
+                    var saveFile = "ZividGemTransformed.zdf";
+                    Console.WriteLine("Saving frame to file: " + saveFile);
+                    frame.Save(saveFile);
```

Comment:
Should we add a comment that calling transform() on pointCloud actually changes the frame?

Reply:
I don't have a strong opinion on that. In my opinion we do not to explain all the api calls in samples, we should only do it if it's of some particular interest in a particular sample. Whether that applies here, I'm not too sure about, I think that's more up to what you want to communicate with this sample.

Reply:
I would not expect that calling transform would modify the frame. However, the sample does show exactly that and the file name indicates it so maybe we don't need it.

Reply:
> I would not expect that calling transform would modify the frame.

I too prefer functions which don't modify the object, but rather return a new, modified object. But many users probably won't need the original non-transformed point cloud again, and I think (didn't check) in-place transforms can be done slightly faster. In my opinion, it would be nice if we had both an in-place `transform()` and a `transformed() const` which returns a new point cloud.